Source/FluentSpreadsheets/Extensions/ComponentBorderExtensions.cs

```diff
@@ -1,75 +1,76 @@
 using System.Drawing;
 using FluentSpreadsheets.Styles;
+using FluentSpreadsheets.Wrappables;
 
 namespace FluentSpreadsheets;
 
 public static class ComponentBorderExtensions
 {
-    public static IComponent WithLeadingBorder(this IComponent component, BorderStyle borderStyle)
+    public static T WithLeadingBorder<T>(this T component, BorderStyle borderStyle) where T : IWrappable<T>
     {
         var frameStyle = new FrameBorderStyle { Leading = borderStyle };
         var style = frameStyle.AsStyle();
 
-        return component.WrapInStyle(style);
+        return component.WithStyleApplied(style);
     }
 
-    public static IComponent WithLeadingBorder(this IComponent component, BorderType? borderType, Color? color)
+    public static T WithLeadingBorder<T>(this T component, BorderType? borderType, Color? color) where T : IWrappable<T>
         => component.WithLeadingBorder(new BorderStyle(borderType, color));
 
-    public static IComponent WithLeadingBorderType(this IComponent component, BorderType borderType)
+    public static T WithLeadingBorderType<T>(this T component, BorderType borderType) where T : IWrappable<T>
         => component.WithLeadingBorder(borderType, null);
```

Comment:
I have no idea how I missed a lot of extension files at previoues reviews but I don't think code with whole lot of the, especially with constraints looks good.

Reply:
Sadly, the comment above doesn't make much sense for me) can you rewrite it, so it would clearer what you mean?

Reply:
Well, I said that having 8 extension files with core logic is in my opinion not really a good structure for the project

Reply:
Why though? 
First of all, any logic that defined within a core project/package is a core logic, if project/package responsibilities are distributed right, which in  this case - they are, doesn't matter what shape that logic has, as long as it makes sense from domain perspective and does not hurt the design. 

Having these methods is beneficial from logic reusability point. We have an interface, that gives us the ability to style and apply modifiers to objects that know how to style and apply modifiers to themselves, the logic of applying styles and modifications to different objects that support this always the same, it is the way that objects receive these operations is different. 

So 1) logic of styles and modifications is not really tied to object themselves, thus the usage of extension methods, 2) other way that comes into my mind is to move these methods to interface itself, which is bad, as you introduce a responsibility to an interface which is not really its concern (OCP hurt really bad as well), and adding insult to injury, target runtime does not support default interface implementations, so you'll either would need to add a base class to all types that implement that interface, which will tie them to unnecessary hierarchy, it is always bad from design perspective, or, even worse, you'll have to duplicate implementations of all these methods in every base class. 

So, in this case, usage of extension methods gives a versatility of sharing a logic across a range of types that conform to this interface (components, component sources, row components).

Reply:
Well, thanks for the essay, I've got absolutely decimated